tests/unit_tests/data_steward/gcloud/bq/bq_test.py

```diff
@@ -40,6 +45,109 @@ def _get_all_field_types(self,) -> typing.FrozenSet[str]:
                             all_field_types.add(field.get('type'))
         return frozenset(all_field_types)
 
+    def list_item_from_table_id(self, table_id: str) -> TableListItem:
+        """
+        Get a table list item as returned by :meth:`bigquery.Client.list_tables` 
+        from a table ID
+        
+        :param table_id: A table ID including project ID, dataset ID, and table ID, 
+        each separated by ``.``. 
+        :return: a table list item
+        """
+        resource = {
+            "tableReference":
+                TableReference.from_string(table_id).to_api_repr()
+        }
+        return TableListItem(resource)
+
+    def mock_query_result(self,
+                          rows: List[Union[Dict, OrderedDict]],
+                          key_order: Iterable[Any] = None):
+        """
+        Create a mock RowIterator as returned by :meth:`bigquery.QueryJob.result`
+        from rows represented as a list of dictionaries
+
+        :param rows: A list of dictionaries representing result rows
+        :param key_order: If `rows` refers to a list of dict rather than OrderedDict, 
+            specifies how fields are ordered in the result schema. This parameter is 
+            ignored if `rows` refers to a list of OrderedDict.
+        :return: a mock RowIterator
+        """
+        mock_row_iter = MagicMock(spec=bigquery.table.RowIterator)
+        mock_row_iter.total_rows = len(rows)
+        row0 = rows[0]
+        if isinstance(row0, OrderedDict):
+            _rows = rows
+        else:
+            _rows = [self._to_ordered_dict(row, key_order) for row in rows]
+            row0 = _rows[0]
+
+        mock_row_iter.schema = list(self.fields_from_dict(row0))
+        field_to_index = {key: i for i, key in enumerate(row0.keys())}
+        mock_row_iter.__iter__ = Mock(return_value=iter([
+            bigquery.table.Row(list(row.values()), field_to_index)
+            for row in _rows
+        ]))
+        return mock_row_iter
+
+    def _to_ordered_dict(self, d: dict, key_order: Iterable[Any] = None):
+        """
+        Convert a dict to OrderedDict with a specified order
+        :param d: instance to convert
+        :param key_order: specifies how items are ordered in the result
+        :return: the ordered dict
+        """
+        if len(d) == 1:
+            return OrderedDict(d)
+        else:
+            if key_order is None:
+                raise ValueError(
+                    'Parameter key_order is required in order to convert'
+                    ' a dict with multiple items to OrderedDict')
+        return OrderedDict((key, d[key]) for key in key_order)
```

Comment:
This seems very specific. If it is not used much (maybe 1/2 instances), I'd suggest removing this function and performing this where it is used. This is not really relevant to the BQ object.